include/circt/Dialect/HW/HWOpInterfaces.td

```diff
@@ -138,6 +138,11 @@ def HWInstanceLike : OpInterface<"HWInstanceLike"> {
     "::mlir::StringAttr", "referencedModuleNameAttr", (ins),
     /*methodBody=*/[{}],
     /*defaultImplementation=*/[{ return $_op.getModuleNameAttr().getAttr(); }]>,
+
+    InterfaceMethod<"Get the referenced module",
+    "::mlir::Operation *", "getReferencedModule", (ins),
+    /*methodBody=*/[{}],
+    /*defaultImplementation=*/[{}]>,
```

Comment:
Random thought: does MLIR support making the `HWInstanceLike` interface parametric over the referenced module op type? That could allow us to more strictly enforce that the instantiated ops be e.g. `FModuleLike` for FIRRTL instances.

Reply:
Discussed offline: we cannot make this HWModuleLike due to MLIR emission ordering. We also want this to be as generic as possible.